Sources/Core/ImagePipeline.swift

```diff
@@ -138,6 +138,33 @@ public final class ImagePipeline {
     ) -> ImageTask {
         loadImage(with: request.asImageRequest(), isConfined: false, queue: queue, progress: progress, completion: completion)
     }
+    
+#if swift(>=5.5)
+    /// Loads an image for the given request.
+    ///
+    /// See [Nuke Docs](https://kean.blog/nuke/guides/image-pipeline) to learn more.
+    ///
+    /// - parameter request: An image request.
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    public func loadImage(with request: ImageRequestConvertible) async throws -> ImageResponse {
+        final class TaskBox {
+            // TODO: Does it need thread-safety?
+            // TODO: Are we sure loadImage will call again every time you await?
+            var task: ImageTask?
+        }
+        let box = TaskBox()
+        return try await withTaskCancellationHandler(handler: {
+            // TODO: How can we report cancellation from here? Add cancellation to ImageTask?
+            box.task?.cancel()
+        }, operation: {
+            try await withUnsafeThrowingContinuation { continuation in
+                box.task = self.loadImage(with: request, queue: nil, progress: nil) {
+                    continuation.resume(with: $0)
+                }
+            }
```

Comment:
Ideally, you would make `ImageTask` conform to `Sendable` and then use it like this:

```suggestion
        var task: ImageTask?

        return try await withTaskCancellationHandler(handler: { [task] in
            // TODO: How can we report cancellation from here? Add cancellation to ImageTask?
            task?.cancel()
        }, operation: {
            try await withUnsafeThrowingContinuation { continuation in
                task = self.loadImage(with: request, queue: nil, progress: nil) {
                    continuation.resume(with: $0)
                }
            }
```

Reply:
That's a good idea, thanks! Initially, I just went with what the compiler suggested. I'm planning to wrap this up today.

Reply:
It doesn't seem that I'll be able to make `ImageTask` `Sendable`. It's not thread-safe and has to be thread-confined. Cancellation is.